zerver/lib/home.py

```diff
@@ -132,12 +133,17 @@ def build_page_params_for_home_page_load(
 
     if user_profile is None:
         request_language = request.COOKIES.get(settings.LANGUAGE_COOKIE_NAME, default_language)
+        split_url = urlsplit(request.build_absolute_uri())
+        show_try_zulip_modal = (
+            settings.DEVELOPMENT or split_url.hostname == "chat.zulip.org"
+        ) and split_url.query == "show_try_zulip_modal"
```

Comment:
I'm not sure we need a conditional on DEVELOPMENT or hostname here; we don't expect there to be a reason anyone would pass `?show_try_zulip_modal` on other servers.

Reply:
But it'd be pretty confusing if shown on a random other server, and the check is harmless, so maybe better to have it than not.